waspc/data/Generator/templates/server/src/middleware.ts

```diff
@@ -0,0 +1,41 @@
+{{={= =}=}}
+import express from 'express'
+import cookieParser from 'cookie-parser'
+import logger from 'morgan'
+import cors from 'cors'
+import helmet from 'helmet'
+
+import config from './config.js'
+
+{=# globalMiddlewareConfigFnDefined =}
+{=& globalMiddlewareConfigFnImportStatement =}
+{=/ globalMiddlewareConfigFnDefined =}
+{=^ globalMiddlewareConfigFnDefined =}
+const {=& globalMiddlewareConfigFnImportAlias =} = (m: MiddlewareConfig) => m
+{=/ globalMiddlewareConfigFnDefined =}
+
+export type MiddlewareConfig = Map<string, express.RequestHandler>
+
+export type MiddlewareConfigFn = (middleware: MiddlewareConfig) => MiddlewareConfig
+
+const _defaultMiddleware: MiddlewareConfig = new Map([
+  ['helmet', helmet()],
+  ['cors', cors({ origin: config.allowedCORSOrigins })],
+  ['logger', logger('dev')],
+  ['express.json', express.json()],
+  ['express.urlencoded', express.urlencoded({ extended: false })],
+  ['cookieParser', cookieParser()]
+])
+
+const defaultMiddleware = {=& globalMiddlewareConfigFnImportAlias =}(_defaultMiddleware)
```

Comment:
I am having a bit of trouble understanding how is this working, what is this `globalMiddlewareConfigFnImportAlias`?

Reply:
Ah ok I kind of get it now! For some reason I found it pretty tricky to understand this file.
Do you think we could somehow make it easier to grasp? I found it somewhat hard to understand the relationships / what is important what is not.
Maybe some comments, different naming, reordering stuff or similar could help?

Reply:
Tried to clean this up and comment a bit 👍🏻